src/backends/github/API.js

```diff
@@ -156,18 +156,35 @@ export default class API {
   }
 
   readFile(path, sha, branch = this.branch) {
-    const cache = sha ? LocalForage.getItem(`gh.${ sha }`) : Promise.resolve(null);
-    return cache.then((cached) => {
-      if (cached) { return cached; }
-
+    if (sha) {
+      return this.getBlob(sha);
```

Comment:
Should we fetch the file by path if the SHA is invalid?

Reply:
Using the Contents API as a backup makes sense. I feel like there are edge cases like file name changes where this could cause confusion, but that seems pretty low risk.

Reply:
I'll need to go back and check -- I'm assuming that we can just check for a 404?

Reply:
Although GitHub doesn't seem to ever do GC -- an invalid blob would be really weird. It would likely mean we were hashing the file differently than GitHub was on the server-side.